### PR TITLE
import: recognize $(EXIF_xxx) in default metadata

### LINKED ISSUE #2452: Use exif variables in metadata fields for retrieving date, etc. when importing files
**Is your feature request related to a problem? Please describe.**
Allow exif variables into the metadata fields at import

**Describe the solution you'd like**
Be able to type something like "Copyright $year J.-Luc" into one of the fields, then dt automatically replaces $year by the corresponding value read from the currently being imported file.

**Describe alternatives you've considered**
Filling the field manually, but this can be a probllem for files that were taken long ago, and one can't remember at which date. Currently the only option.

**Additional context**

### PR DESCRIPTION
**Pull request by phweyland**
fixes #2452

### CODE REVIEW

File: src/common/variables.c [hunk 1]
```diff
@@ -139,7 +140,7 @@ static void init_expansion(dt_variables_params_t *params, gboolean iterate)
 
     params->data->flags = img->flags;
 
-    dt_image_cache_read_release(darktable.image_cache, img);
+    if(params->img == NULL) dt_image_cache_read_release(darktable.image_cache, img);
```
Comment (TurboGit) [inline src/common/variables.c:143]: And when is params->img released if set ?

Comment (phweyland) [reply src/common/variables.c:143]: My mistake ... :cry:

Comment (phweyland) [reply src/common/variables.c:143]: Sorry, thinking better. For me that's correct. 
params->img set, means that the cache is already got, in that case in _import_image_internal().
So, for this time (dt_exif_apply_default_metadata()), init_expansion, doesn't need not to get it, neither to release it.

Comment (TurboGit) [reply src/common/variables.c:143]: I see now, thanks.

**Comment by TurboGit** [comment]
Manually merged on master and queued for 3.4.1.

**Pull request closed without merging**